geomstats/geometry/stratified/graph_space.py

```diff
@@ -381,141 +571,474 @@ def dist(self, graph_a, graph_b):
+        metric : GraphSpaceMetric
+        base_graph : array-like, shape=[..., n_nodes, n_nodes]
             Base graph.
-        graph_to_permute : array-like, shape=[m, n, n]
+        graph_to_permute : array-like, shape=[..., n_nodes, n_nodes]
             Graph to align.
 
         Returns
         -------
-        permutation : array-like, shape=[m,n]
-            Node permutation indices of the second graph.
+        permuted_graph : array-like, shape=[..., n_nodes, n_nodes]
+            Permuted graph as to be aligned with respect to the geodesic.
         """
-        return [
+        base_graph, graph_to_permute, is_single = self._broadcast(
+            base_graph, graph_to_permute
+        )
+
+        perm = [
             gs.linalg.quadratic_assignment(x, y, options={"maximize": True})
             for x, y in zip(base_graph, graph_to_permute)
         ]
 
+        self.perm_ = gs.array(perm[0]) if is_single else gs.array(perm)
+
+        return self._permute(metric, graph_to_permute, self.perm_)
 
-class IDMatcher(_Matcher):
-    """Identity matching."""
 
-    def match(self, base_graph, graph_to_permute):
-        """Match graphs.
+class IDAligner(_BaseAligner):
+    """Identity alignment.
+
+    The identity alignment is not performing any matching but returning the nodes in
+    their original position. This alignment can be selected when working with labelled
+    graphs.
+    """
+
+    def align(self, metric, base_graph, graph_to_permute):
+        """Align graphs.
 
         Parameters
         ----------
-        base_graph : array-like, shape=[m, n, n]
+        metric : GraphSpaceMetric
+        base_graph : array-like, shape=[..., n_nodes, n_nodes]
             Base graph.
-        graph_to_permute : array-like, shape=[m, n, n]
+        graph_to_permute : array-like, shape=[..., n_nodes, n_nodes]
             Graph to align.
 
         Returns
         -------
-        permutation : array-like, shape=[m,n]
-            Node permutation indices of the second graph.
+        permuted_graph : array-like, shape=[..., n_nodes, n_nodes]
+            Permuted graph as to be aligned with respect to the geodesic.
         """
+        base_graph, graph_to_permute, is_single = self._broadcast(
+            base_graph, graph_to_permute
+        )
+
         n_nodes = base_graph.shape[1]
-        return gs.reshape(gs.tile(range(n_nodes), base_graph.shape[0]), (-1, n_nodes))
+        perm = gs.reshape(gs.tile(range(n_nodes), base_graph.shape[0]), (-1, n_nodes))
+
+        self.perm_ = gs.array(perm[0]) if is_single else gs.array(perm)
+
+        return self._permute(metric, graph_to_permute, self.perm_)
+
+
+class ExhaustiveAligner(_BaseAligner):
+    """Brute force exact alignment.
+
+    Exact Alignment obtained by exploring the whole permutation group.
+
+    Notes
+    -----
+    Not recommended for large `n_nodes`.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._all_perms = None
```

Comment:
The convention was to put the _ at the end of the variable's name, in the other classes: any reason for the change here?

Reply:
This is just a private attribute of the class.